client/question/components/LearnerViewDirective.js

```diff
@@ -25,79 +25,91 @@ tie.directive('learnerView', [function() {
         <div class="tie-question-ui-outer">
           <div class="tie-question-ui-inner">
             <div class="tie-question-ui">
-              <div class="tie-question-window" ng-show="!MonospaceDisplayModalService.isDisplayed()">
-                <div class="tie-question-container">
+              <div class="tie-question-window"
+                  ng-show="!MonospaceDisplayModalService.isDisplayed()">
+                <div class="tie-question-container" ng-class="{'pulse-animation-enabled': pulseAnimationEnabled}">
                   <h3 class="tie-question-title">{{title}}</h3>
                   <div class="tie-previous-instructions">
                     <div ng-repeat="previousInstruction in previousInstructions track by $index">
                       <div ng-repeat="instruction in previousInstruction track by $index">
-                        <p ng-if="instruction.type == 'text'">{{instruction.content}}</p>
+                        <p ng-if="instruction.type == 'text'">
+                          {{instruction.content}}
+                        </p>
                         <pre class="tie-question-code" ng-if="instruction.type == 'code'">{{instruction.content}}</pre>
                       </div>
                       <hr>
                     </div>
                   </div>
                   <div class="tie-instructions">
                     <div ng-repeat="instruction in instructions">
-                      <p ng-if="instruction.type == 'text'">{{instruction.content}}</p>
+                      <p ng-if="instruction.type == 'text'">
+                        {{instruction.content}}
+                      </p>
                       <pre class="tie-question-code" ng-if="instruction.type == 'code'">{{instruction.content}}</pre>
                     </div>
                   </div>
+                  <div class="tie-dot-container" ng-class="{'tie-display-dots': ConversationLogDataService.isNewBalloonPending()}"}>
```

Comment:
Why can't this go in the speech balloons container directive? Whether a new balloon is pending seems to be a property of that container, it's a bit odd to split it like this. The exception would be if something outside that container triggers the dots.

That way you also don't need to expose $scope.ConversationLogDataService in its entirety here.

Reply:
It was easier to structure the UI this way to get it to behave the way it does, I didn't feel the dots are the same thing as the balloons (the broader feedback window is waiting for the balloons to respond, so I view the dots as part of the window containing the balloons), and ConversationLogDataService seems to mostly be exposed anyway.

Reply:
I'm not sure why structuring the UI this way is easier -- couldn't you just put the loading dots at the start of the child directive, or does this interfere with the animation somehow? Also, I agree that the dots aren't the same things as the balloons, but the presence of the loading dots depends only on the speech bubbles themselves -- so I'd like to strongly suggest that they stay in the child directive, since all the stuff they depend on is in there as well.

ConversationLogDataService being exposed is a temporary thing; I'm planning to find a pattern that allows retrieval without direct access to properties on the service. I think I have an idea for how to do this but will defer it to a future PR -- but in the meantime let's not propagate its exposure.

Reply:
Done